string_helpers.py

```diff
@@ -1,29 +1,41 @@
-###############################################################
-def string_used_words(used_letters): # Crea lista de letras usadas con espacio en el medio
+def show_used_letters(used_letters):
+    """
+    Converts list to str and concatenates ' '
+    """ 
     result = ""
     for x in used_letters:
         result += x.upper() + " "
     return result
 
-###############################################################
-def busca_indices(letter, hidden_word, graphic_word): # Busca matches entre letter y hidden_word
-    graphic_word = list(graphic_word)                # Devuelve graficword converitdo
-    for index, x in enumerate(hidden_word):  # genera en index el indice, 
-#                                            # y en x el objeto dentro de hidden_word
+###########################################################
+
+def search_matches(letter, hidden_word, graphic_word): 
+    """
+    Searches letter matches in hidden word
+    Returns graphic_word updated  
+    """
+    
+    graphic_word = list(graphic_word)                
+    for index, x in enumerate(hidden_word):  
         if x == letter:
             graphic_word[index] = letter
     return graphic_word
+
 ################################################################
 
-def convierte_string(graphic_word):   #convierte en str a graphic word, ya q se convierte en lista antes
-    graphic_word = list(graphic_word)
+def round_begins(graphic_word, hidden_word):
+    """
+    Makes graphic_word: reveals 1st and last letters, fullfits string with '.'
+    """ 
+    graphic_word = hidden_word[0] + ("." * (len(hidden_word) - 2)) + hidden_word[-1]
+    graphic_word = search_matches(graphic_word[0], hidden_word, graphic_word) 
+    graphic_word = search_matches(graphic_word[-1], hidden_word, graphic_word)
     graphic_word[0] = graphic_word[0].upper()
     graphic_word[-1] = graphic_word[-1].upper()
-    return "".join(graphic_word)
-###########################################################
+    return "".join(graphic_word) 
 
-def round_begins(graphic_word, hidden_word): # crea grafic_word con '.' en el medio 
-    graphic_word = hidden_word[0] + ("." * (len(hidden_word) - 2)) + hidden_word[-1]
-    graphic_word = busca_indices(graphic_word[0], hidden_word, graphic_word) 
-    graphic_word = busca_indices(graphic_word[-1], hidden_word, graphic_word)
-    return graphic_word                      # no era funcion porque al princio solo se jugaba una vuelta
+def result_table(score_list):
+    result = ""
+    for x in score_list:
+        result += str(x) + '\n'
+    return result
```

Comment:
the python way:

Todo este codigo se puede cambiar por

```python
"\n".join(score_list)
```

Esto funciona solo si todos los elementos son string. Para podes usar una funcion `map` junto con `str`. 

```python
map(str, score_list)  # Esto recorre la lista y por cada elemento aplica la funcion que pasaste en el primer argumento. Te va a devolver basicamente una nueva lista con todos los elementos convertidos a string
```

Asi que si queres asegurarte de que siempre sea todo string, podes hacer esta linea:

```python
"\n".join(map(str, score_list))
```